flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -471,20 +471,25 @@ class ScalarExprLowering {
     TODO(getLoc(), "implied do index");
   }
 
+  // FIXME: Assumption is these return an i64 regardless of the target.
```

Comment:
There is `Fortran::evaluate::DescriptorInquiry::Result` that specifies the result type defined by the front-end for descriptor inquiries.
So I think you can do something like:
```
using ResultT = Fortran::evaluate::DescriptorInquiry::Result;
auto type = converter.genType(ResultT::category, ResultT::kind);
```
To get the result type defined by the front-end without assuming what it is here.

Reply:
I'll give that a go. Thanks, Jean.